src/main/java/org/linksharing/server/auth/AuthController.java

```diff
@@ -47,7 +51,12 @@ public String registerUser(@Valid User user, BindingResult result) {
             return "redirect:/login?userExists";
         }
 
+        UserProfileDetails userProfile = new UserProfileDetails();
+        userProfile.setEmail(user.getEmail());
+        userProfile.setFirstName(user.getUsername());
```

Comment:
Let's use the username for user's public url (the url when a user sees the profile details and links, the url to share). For example, http://localhost:8080/u/<username>.

So, first name is not username.

Reply:
Ok, I'll save it for tomorrow

Reply:
Don't quite understand logic. The URL   http://localhost:8080/u/username   should be unique for each user. So username should be unique for each user also, as it's part of URL. But it's not, cause we didn't provide that.
Or maybe I don't get? What is /u/ in URL ?

Reply:
Yes, you're right. See #12